lib/ansible/modules/packaging/language/gem.py

```diff
@@ -252,6 +272,8 @@ def main():
         module.fail_json(msg="Cannot specify version when state=latest")
     if module.params['gem_source'] and module.params['state'] == 'latest':
         module.fail_json(msg="Cannot maintain state=latest when installing from local source")
+    if module.params['user_install'] and module.params['install_dir']:
+        module.fail_json(msg="install_dir requires user_install=false")
```

Comment:
Rather than adding an explicit check, you could just set `user_install` and `install_dir` as mutually exclusive.

Reply:
Tell me if I'm wrong. But I'm not sure this is possible.

In order for this feature to work, the user MUST specify:

```yaml
gem:
  user_install: no
  install_dir: /some/other/directory
```

The default of `user_install` being `True`.

Unless I'm mistaking, `mutually_exclusive` means that "the user CAN specify
one, but MUST NOT specify the other." However, here, the user MUST specify both.

If I'm right, there are two solution:

  * Keep the current code (= the manual check)
  * Change the default of `user_install` to `no`, thus loosing backward compatibility.